lib/src/main/java/uk/gov/di/ipv/core/library/service/UserIdentityService.java

```diff
@@ -91,6 +108,74 @@ public Map<String, String> getUserIssuedDebugCredentials(String ipvSessionId) {
         return userIssuedDebugCredentials;
     }
 
+    private String generateVectorOfTrustClaim(
+            List<UserIssuedCredentialsItem> credentialIssuerItems) {
+        Optional<Boolean> validPassport = Optional.empty();
+        Optional<Boolean> validFraud = Optional.empty();
+        Optional<Boolean> validKbv = Optional.empty();
+
+        for (UserIssuedCredentialsItem item : credentialIssuerItems) {
+            try {
+                JWTClaimsSet jwtClaimsSet = SignedJWT.parse(item.getCredential()).getJWTClaimsSet();
+                JSONObject vcClaim = (JSONObject) jwtClaimsSet.getClaim(VC_CLAIM);
+                JSONArray evidenceArray = ((JSONArray) vcClaim.get(VC_EVIDENCE));
+
+                if (item.getCredentialIssuer().equals(PASSPORT_CRI_TYPE)) {
+                    validPassport =
+                            isValidScore(
+                                    evidenceArray,
+                                    GPG_45_VALIDITY_PROPERTY_NAME,
+                                    GPG_45_MAX_VALIDITY_SCORE);
+                }
+
+                if (item.getCredentialIssuer().equals(FRAUD_CRI_TYPE)) {
+                    validFraud =
+                            isValidScore(
+                                    evidenceArray,
+                                    GPG_45_FRAUD_PROPERTY_NAME,
+                                    GPG_45_MAX_FRAUD_SCORE);
+                }
+
+                if (item.getCredentialIssuer().equals(KBV_CRI_TYPE)) {
+                    validKbv =
+                            isValidScore(
+                                    evidenceArray,
+                                    GPG_45_VERIFICATION_PROPERTY_NAME,
+                                    GPG_45_MAX_VERIFICATION_SCORE);
+                }
+            } catch (ParseException e) {
+                LOGGER.warn("Failed to parse VC JWT");
+            }
+        }
+
+        return getVectorOfTrustValue(validPassport, validFraud, validKbv);
+    }
+
+    private Optional<Boolean> isValidScore(
+            JSONArray evidenceArray, String property, int scoreValue) {
+        Long gpg45ScoreValue = ((Map<String, Long>) evidenceArray.get(0)).get(property);
+        if (gpg45ScoreValue != null) {
+            return Optional.of(gpg45ScoreValue.intValue() == scoreValue);
```

Comment:
I know we won't be receiving it for MVP, but using passport as an example, it's possible to get a validity score of higher than 2. Should this logic be `gpg45ScoreValue.intValue() >= scoreValue`? And perhaps `GPG_45_MAX_VALIDITY_SCORE ` might be called something like `GPG_45_M1A_VALIDITY_SCORE `. Or something. I'm hazy on gpg45 details so I may be way off the mark here.